File: Converter/app/src/main/java/com/example/converter/MainActivity.java
package com.example.converter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button buttonmtokm = (Button) findViewById(R.id.buttonmtokm);
        buttonmtokm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                EditText textBoxMiles = (EditText) findViewById(R.id.editm);
                EditText textBoxKm = (EditText) findViewById(R.id.editkm);
                double vMiles = Double.valueOf(textBoxMiles.getText().toString());
                double vKm = (vMiles /0.62137);
                DecimalFormat formatVal = new DecimalFormat("##.##");
                textBoxKm.setText(formatVal.format(vKm));
            }
        });
        Button buttonkmtom = (Button) findViewById(R.id.buttonkmtom);
        buttonkmtom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                EditText textBoxMiles = (EditText) findViewById(R.id.editm);
                EditText textBoxKm = (EditText) findViewById(R.id.editkm);
                double vKm = Double.valueOf(textBoxKm.getText().toString());
                double vMiles = (vKm *0.621371);
                DecimalFormat formatVal = new DecimalFormat("##.##");
                textBoxMiles.setText(formatVal.format(vMiles));
            }
        });
    }
}

File: MAD_LAB/Parser/app/src/main/java/csmp/part_a/p6/activities/MainActivity.kt
package csmp.part_a.p6.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import csmp.part_a.p6.R

class MainActivity : AppCompatActivity(), View.OnClickListener {
    var parseXmlButton: Button? = null
    var parseJsonButton: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        parseJsonButton = findViewById<View>(R.id.parseJsonButton) as Button
        parseXmlButton = findViewById<View>(R.id.parseXmlButton) as Button
        parseJsonButton!!.setOnClickListener(this)
        parseXmlButton!!.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.parseXmlButton -> {
                val xmlIntent = Intent(this@MainActivity, ParsedXml::class.java)
                startActivity(xmlIntent)
            }
            R.id.parseJsonButton -> {
                val jsonIntent = Intent(this@MainActivity, ParsedJson::class.java)
                startActivity(jsonIntent)
            }
        }
    }
}